File: desktop/src/main/java/com/techfinance/pessoal/desktop/service/TransactionApiService.java
package com.techfinance.pessoal.desktop.service;

import java.util.List;

import com.google.inject.Inject;
import com.google.inject.Singleton;
import com.techfinance.pessoal.desktop.dto.request.TransactionRequest;
import com.techfinance.pessoal.desktop.dto.response.TransactionResponse;
import com.techfinance.pessoal.desktop.infrastructure.http.ApiClient;

@Singleton
public class TransactionApiService {

    private final ApiClient apiClient;

    @Inject
    public TransactionApiService(ApiClient apiClient) {
        this.apiClient = apiClient;
    }

    public List<TransactionResponse> findAll() {
        return apiClient.getList("/transaction", TransactionResponse.class);
    }

    public TransactionResponse create(TransactionRequest request) {
        return apiClient.post("/transaction/create", request, TransactionResponse.class);
    }
}

File: web/src/services/transactionApiService.ts
import { apiClient } from '../infrastructure/http/apiClient'
import type {
  TransactionRequest,
  TransactionResponse,
} from '../types/transaction'

class TransactionApiService {
  findAll(): Promise<TransactionResponse[]> {
    return apiClient.getList<TransactionResponse>('/transaction')
  }

  create(request: TransactionRequest): Promise<TransactionResponse> {
    return apiClient.post<TransactionResponse>('/transaction/create', request)
  }
}

export const transactionApiService = new TransactionApiService()
